common/src/common/provider-get.ts

```diff
@@ -31,8 +33,14 @@ const providerGetWindow = async (chainId = 0): Promise<Web3Provider | undefined>
 
 const providerGetSigner = async (chainId = 0, accountOrIndex: string | number = 0): Promise<Signer | undefined> => {
   const provider = await providerGetWindow(chainId);
+  if (!provider) return;
 
-  return provider && provider.getSigner(accountOrIndex);
+  const signer = provider.getSigner(accountOrIndex);
+  const signerAddress = (await signer.getAddress()) as Address;
+
+  await receiveFunds(signerAddress, chainId);
```

Comment:
check before chainId need this ("paymaster" attribute in networks.json)

`if hasPaymaster(chainId) {...}`